content/10-minute-tutorial.md

```diff
@@ -1,3 +1,328 @@
+
+Open the file (double click it) and add the feature description, starting with the `Feature` keyword and an optional description.
+
+For instance:
+```gherkin
+Feature: Belly
+  Optional description of the feature
+```
+
+## Creating a Scenario
+
+Scenarios are added to the feature file, to define examples of the expected behaviour. These scenarios can be used to test the feature.
+Start a scenario with the `Scenario` keyword and add a brief description of the scenario. To define the scenario, you have to define all of its steps.
+
+## Defining Steps
+
+These all have a keyword (`Given`, `When`, and `Then`) followed by a step. The step is then matched to a [step definition](/step-definitions/) which map the plain text step to programming code.
+
+The plain text steps are defined in the [Gherkin](/gherkin/) language. Gherkin allows developers and business stakeholders to describe and share the expected behaviour of the application. It should not describe the implementation.
+
+The feature file contains the Gherkin source.
+
+- The `Given` keyword precedes text defining the context; the known state of the system (or precondition).
+
+- The `When` keyword precedes text defining an action.
+
+- The `Then` keyword precedes text defining the result of the action on the context (or expected result).
+
+For instance:
+```gherkin
+  Scenario: a few cukes
+    Given I have 42 cukes in my belly
+    When I wait 1 hour
+    Then my belly should growl
+```
+
+# Running the test
+
+To run the tests from JUnit we need to add a runner to our project.
+
+Create a new Java class in your `src/test/java/<project>` directory, called `RunCucumberTest.java`.
+
+```java
+package <project>;
+
+import cucumber.api.CucumberOptions;
+import cucumber.api.junit.Cucumber;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(plugin = {"pretty"})
+public class RunCukesTest {
+}
+```
+
+The JUnit runner will by default use classpath:package.of.my.runner to look for features
+
+You should get something like the following result:
+```
+1 Scenarios (1 undefined)
+3 Steps (3 undefined)
+0m0.015s
+
+
+You can implement missing steps with the snippets below:
+
+@Given("^I have (\\d+) cukes in my belly$")
+public void i_have_cukes_in_my_belly(int arg1) throws Exception {
+    // Write code here that turns the phrase above into concrete actions
+    throw new PendingException();
+}
+
+@When("^I wait (\\d+) hour$")
+public void i_wait_hour(int arg1) throws Exception {
+    // Write code here that turns the phrase above into concrete actions
+    throw new PendingException();
+}
+
+@Then("^my belly should growl$")
+public void my_belly_should_growl() throws Exception {
+    // Write code here that turns the phrase above into concrete actions
+    throw new PendingException();
+}
+
+
+Process finished with exit code 0
```

Comment:
Myabe add a little explanation. here of the step implementations shown? Or just a bit of flow text that takes us into the next section.

Reply:
Added